src/protocols/secure_channel/CASESession.cpp

```diff
@@ -335,54 +348,51 @@ CHIP_ERROR CASESession::DeriveSecureSession(SecureSession & session, SecureSessi
 
 CHIP_ERROR CASESession::SendSigmaR1()
 {
-    uint16_t data_len =
-        static_cast<uint16_t>(kSigmaParamRandomNumberSize + sizeof(uint16_t) + sizeof(uint16_t) +
-                              mOpCredSet->GetCertCount() * kTrustedRootIdSize + kP256_PublicKey_Length + sizeof(uint64_t) * 4);
+    uint16_t data_len = static_cast<uint16_t>(kSigmaParamRandomNumberSize + sizeof(uint16_t) + kMAX_Hash_Length +
+                                              kP256_PublicKey_Length + sizeof(uint64_t) * 4);
 
     System::PacketBufferTLVWriter tlvWriter;
     System::PacketBufferHandle msg_R1;
     TLV::TLVType outerContainerType                      = TLV::kTLVType_NotSpecified;
     uint8_t initiatorRandom[kSigmaParamRandomNumberSize] = { 0 };
-
-    msg_R1 = System::PacketBufferHandle::New(data_len);
-    VerifyOrReturnError(!msg_R1.IsNull(), CHIP_ERROR_NO_MEMORY);
+    uint8_t destinationIdentifier[kMAX_Hash_Length]      = { 0 };
 
     // Step 1
-    // Fill in the random value
+    ReturnErrorOnFailure(mEphemeralKey.Initialize());
+
+    // Step 2
+    {
+        MutableByteSpan ipkSpan(mIPK, sizeof(mIPK));
+        ReturnErrorOnFailure(ComputeIPK(mConnectionState.GetLocalKeyID(), ipkSpan));
+    }
+
+    // Step 3
     ReturnErrorOnFailure(DRBG_get_bytes(initiatorRandom, kSigmaParamRandomNumberSize));
 
-    // Step 4
-    ReturnErrorOnFailure(mEphemeralKey.Initialize());
+    // Step 6
```

Comment:
These step numbers are confusing: step 6 is before `Step 5 (6)`. Would it be possible instead to describe, rather than number, the steps?

Reply:
Changed the comments to a description rather than step numbers.